docs/DEVELOPER_GUIDE.md

```diff
@@ -1 +1,179 @@
+   - [Azure Container Registry quickstart](https://docs.microsoft.com/en-us/azure/container-registry/container-registry-get-started-portal)
+- **Note**: if you are using docker hub to store your images your
+  `KO_DOCKER_REPO` variable should be `docker.io/<username>`.
+- **Note**: Currently Docker Hub doesn't let you create subdirs under your
+  username.
+
+`.bashrc` example:
+
+```shell
+export GOPATH="$HOME/go"
+export PATH="${PATH}:${GOPATH}/bin"
+export KO_DOCKER_REPO='docker.io/<username>'
+```
+
+### Checkout your fork
+
+The Go tools require that you clone the repository to the
+`src/github.com/kubeflow/kfserving` directory in your
+[`GOPATH`](https://github.com/golang/go/wiki/SettingGOPATH).
+
+To check out this repository:
+
+1. Create your own
+   [fork of this repo](https://help.github.com/articles/fork-a-repo/)
+1. Clone it to your machine:
+
+```shell
+mkdir -p ${GOPATH}/src/github.com/kubeflow
+cd ${GOPATH}/src/github.com/kubeflow
+git clone git@github.com:${YOUR_GITHUB_USERNAME}/kfserving.git
+cd serving
+git remote add upstream git@github.com:kubeflow/kfserving.git
+git remote set-url --push upstream no_push
+```
+
+_Adding the `upstream` remote sets you up nicely for regularly
+[syncing your fork](https://help.github.com/articles/syncing-a-fork/)._
+
+Once you reach this point you are ready to do a full build and deploy as
+described below.
+
+## Deploy KFServing
+
+### Check Knative Serving installation
+Once you've [setup your development environment](#prerequisites), you can see things running with:
+
+```console
+kubectl -n knative-serving get pods
+NAME                          READY     STATUS    RESTARTS   AGE
+activator-c8495dc9-z7xpz      2/2       Running   0          6d
+autoscaler-66897845df-t5cwg   2/2       Running   0          6d
+controller-699fb46bb5-xhlkg   1/1       Running   0          6d
+webhook-76b87b8459-tzj6r      1/1       Running   0          6d
+```
+### Deploy KFServing from default
+
+```bash
+make deploy
+```
+
+After above step you can see things running with:
+```console
+kubectl get pods -n kubeflow-system -l control-plane=kfserving-controller-manager
+NAME                             READY   STATUS    RESTARTS   AGE
+kfserving-controller-manager-0   2/2     Running   0          13m
+```
+- **Note**: By default it installs to `kubeflow-system` namespace with the published
+`kfserving-controller-manager` image.
+
+### Deploy KFServing with your own version
+```bash
+make deploy-test
+```
+- **Note**: `deploy-test` builds the image from your local code, publishes to `KO_DOCKER_REPO`
+and deploys the `kfserving-controller-manager` with the image digest to your cluster for testing.
+
+
+### Smoke test after deployment
+```bash
+kubectl apply -f docs/samples/tensorflow.yaml
+```
+You should see model serving deployment running under default or your specified namespace.
+
+```console
+kubectl get pods -n default -l serving.knative.dev/configuration=my-model-default
```

Comment:
We should 100% label our pods with serving.kubeflow.org/kfservice=my-model

Reply:
ye we can add this